### PR TITLE
feat(apiserver/dockerfile): add es http_auth environments, make docker-compose work in M1 mac and fix nginx bug

### PR DESCRIPTION
**Pull request by weixiao-huang**
This PR does these things:
- add es http_auth environments, which can be used to set http_auth to elasticsearch.
- add `set -o errexit`, `set -o nounset`, `set -o pipefail` into `docker/build/internal_files/final_image_preparation.sh` for making it exit while error in scripts' executing.
- fix a webserver bug for `clearml.conf.template` not having a ended `}`
- use correct mongo and elasticsearch version in `docke-compose.yml` since elasticsearch:7.6.2 and mongo:3.6.5 will cause errors in a M1 mac

### CODE REVIEW

**Comment by jkhenning** [comment]
@weixiao-huang I think the docker-compose volume-related changes make it less clear for linux deployments... maybe we can just create a docker-compose file for M1 mac?

**Comment by weixiao-huang** [comment]
Could we rename current `docker-compose.yaml` to `docker-compose-linux.yaml` and make the volume-related docker-compose file named `docker-compose.yaml`? For I think it will work in Mac, Linux even Windows, which has more compatibility. Compared with clarity and compatibility, maybe the volume-related docker should be named `docker-compose.yaml`.
Another changes I pushed from the latest commit is to make the mongo, es version match in these docker-compose files. I choose a more compatible version that it can work in multiple os system.

**Comment by jkhenning** [comment]
> Could we rename current docker-compose.yaml to docker-compose-linux.yaml and make the volume-related docker-compose file named docker-compose.yaml?

Well, there are two issues with the volumes approach:

1. It's not backwards compatible, and we really try to keep things backwards compatible to make sure ClearML users have the most pleasant upgrade experience... 🙂 (for example, I wouldn't like people who already use ClearML Server and update their docker-compose file to suddenly find the server not starting up since the volumes need to be re-mapped - I'm not even sure I know how to re-map the volumes if the previous setup uses host directories, as it does right now)

2. While it's much easier (🙂), it feels less robust deployment-wise, since there's much less control over the location of the data, and much harder for us to provide instructions on how to backup/handle the data in case of major version migrations

In short, I'd much rather keep the current file naming and add a new docker-compose with some appropriate name to indicate the use of named volumes (or a name that would suggest platform compatibility, for example mac)

> Another changes I pushed from the latest commit is to make the mongo, es version match in these docker-compose files. I choose a more compatible version that it can work in multiple os system.

The version are a great fix 👍  (BTW, in future versions we'll probably upgrade to MongoDB 4.4)

Can you perhaps keep this PR for the Elastic Auth and versions contribution, and open a new PR for the volumes issue? That would be greatly appreciated! 🙏

**Comment by weixiao-huang** [comment]
I got it, maybe a volume-related docker-compose is unnecessary in this situation. I remove these changes and only keep the Elastic Auth and versions contribution.

**Comment by weixiao-huang** [comment]
I finished the codes, please check if there's any problems. Thx

**Comment by jkhenning** [comment]
@weixiao-huang please see my last 5 comments

**Comment by weixiao-huang** [comment]
I'm sorry that I wonder where's the last 5 comments? It seems that I may not find these. Could you please figure it out?

**Comment by jkhenning** [comment]
Just look at the comments in the files: https://github.com/allegroai/clearml-server/pull/93/files 🙂

**Comment by weixiao-huang** [comment]
I could not find any comments in these files
image

Can you see it in your web browser?

**Comment by jkhenning** [comment]
Sorry, my bad 🤣 - it seems I had to submit the review...

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -39,6 +40,7 @@ services:
     container_name: clearml-elastic
     environment:
       ES_JAVA_OPTS: -Xms2g -Xmx2g
+      ELASTIC_PASSWORD: elastic
```
[the remark this answers is not in the document]
Comment (weixiao-huang) [reply docker/docker-compose.yaml:43]: Yes it can work if set empty

**Comment by weixiao-huang** [comment]
I've resolved all comments. Please check it

**Comment by jkhenning** [comment]
Great, just missing `CLEARML_ELASTIC_SERVICE_PASSWORD: ${ELASTIC_PASSWORD}` and `ELASTIC_PASSWORD: ${ELASTIC_PASSWORD}` in the `docker/docker-compose-win10.yml` file 🙂

**Comment by weixiao-huang** [comment]
Done
